src/Platform/Microsoft.Testing.Extensions.Retry/RetryOrchestrator.cs

```diff
@@ -275,7 +275,6 @@ public async Task<int> OrchestrateTestHostExecutionAsync()
                     }
                 }
 
-                finalArguments.Clear();
```

Comment:
Isn't the continue below going back to the next iteration of `while (attemptCount < userMaxRetryCount + 1)` which will start using finalArguments again?

Reply:
As far as I can see we don't use this variable after the foreach on lines 176-183.

Reply:
Sorry I missed the outer loop, you are right this is not correct. I need to refactor this class, there is too much indentation and things to keep in mind.

Reply:
yeah i made the same mistake. and agreed this class is difficult to reason about